### PR TITLE
feat: Configurable Header Logo Link and Footer Links

### PR DESCRIPTION
**Pull request by j0mc**
# Overview

This PR adds an optional link from the header logo to a configurable dashboard URL.
It also adds a configuration option to override the default links in the footer (Privacy Policy, Terms of Service, and Honor Code)

## Test Steps
* Set following configuration options in `.env.private`:
```
DISABLE_APP_FOOTER=false
DISABLE_DESKTOP_HEADER=false
DISABLE_HEADER_LOGO=false
FOOTER_LINK_MAP='[{"label": "Test Link", "url": "https://wgu.edu/", "id": 1}]'
ENABLE_DASHBOARD_RETURN_LINK=true
DASHBOARD_BASE_URL='https://academy.test'
```
* Check that the only footer URL is now "Test Link" and links to the specified URL
* Check that the header links to https://academy.test/course/overview/<course-id>

## Checklist

⚠️ **Please make sure to fill this checklist before asking for reviews.**

- [x] Code is correctly formatted and linted
- [x] Unit tests are updated and are passing
- [x] PR title follows [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary):
      i.e. `feat(CUR-###): Feature title`
- [x] PR Title aligns to Semantic-Release [supported prefixes](https://www.conventionalcommits.org/en/v1.0.0/#examples) (NOTE: prefixes are case sensitive, keep lower-case):

```diff
 feat() - Feature (0.X.0)
 fix() - Patch (0.0.X)
 docs() - Patch (0.0.X), will only scope to README changes
 refactor() - Patch (0.0.X)
 revert() - Patch (0.0.X)
 style() - Patch (0.0.X)
```

- [x] Check for unused files
- [x] Check work before asking for reviewers
- [x] Fix any linting errors
- [x] SECURITY: No secrets where committed to the repo
- [x] COMPLIANCE: Committed code is not proprietary and adheres to Open Source licensing

### CODE REVIEW

File: src/features/footer/Footer.jsx [hunk 1]
```diff
@@ -98,10 +93,16 @@ const Footer = ({ className }) => {
 
 Footer.propTypes = {
   className: PropTypes.string,
+  links: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.oneOfType([PropTypes.string, PropTypes.number]))),
 };
 
 Footer.defaultProps = {
   className: '',
+  links: [
+    { label: 'Privacy Policy', url: getConfig().PRIVACY_POLICY_URL, id: 1 },
```
Comment (rodmgwgu) [inline src/features/footer/Footer.jsx:102]: Not sure if calling getConfig here, outside of the render function, could cause issues:

getConfig values can come from the build-time .env files (which wouldn't have issues here), or from the config in the OpenEDX Database, which come from an async request done during the app initialization (see the APP_READY event on the main index file). Not sure if that event would be ready when this is evaluated.

Comment (j0mc) [reply src/features/footer/Footer.jsx]: Thanks @rodmgwgu , good point. It's probably not good practice to set default props from anything that isn't completely static. 
That being said, I tested in my local environment by building a docker image with this branch and updating `MFE_CONFIG` within the LMS and those settings were picked up correctly from the LMS through `/api/mfe_config/v1?mfe=learning-dev`. It looks like defaultProps is evaluated just before the first render() call for an element.

If this is still a concern maybe we could change Footer to a class based component and set the links in the constructor, or set the defaultProp['links'] = null and pull from the config within the render function if they're not passed in as a prop.

I pushed another possible solution, does this look better, should we stick with what we originally had?

**Review by rodmgwgu — APPROVED** [review]
LGTM, thanks!

File: src/features/footer/Footer.jsx (hunk 1 above)
Comment (rodmgwgu) [reply src/features/footer/Footer.jsx]: the solution looks good to me, thanks!

**Review by holaontiveros — APPROVED** [review]

**Pull request merged**